src/xtgeo/xyz/polygons.py

```diff
@@ -439,7 +440,30 @@ def _name_to_none_if_missing(self):
             self._hname = None
 
     # ----------------------------------------------------------------------------------
-    # Methods
+    # Class methods
+    # ----------------------------------------------------------------------------------
+
+    @classmethod
+    def boundary_from_points(cls, points, alpha: Optional[float] = None):
+        """Instantiate polygons from detecting the boundary around points.
+
+        Args:
+            points: The XTGeo Points instance to estimate boundary/boundaries around.
+            alpha: The alpha factor for determine the 'precision' in how to delineate
+                the polygon. A large value will produce a smoother polygon. The default
+                is to detect the value from the data, but note that this default may be
+                far from optimal for you needs.
+
+        Returns:
+            A Polygons instance.
+
+        .. versionadded: 3.1.0
+        """
+
+        return cls(_polygons_oper.boundary_from_points(points, alpha))
```

Comment:
To make this function more similar to the one used on the regsurf:

- It would be nice if we could have the convex option in this function as well
- Could the `alpha` argument be switched to an `alpha_factor` argument instead? I guess it would require a bit of refactoring. But as long as we can compute the minimum alpha from the pointset, the alpha factor is much easier for the user to play with that the actual alpha.

Reply:
> But as long as we can compute the minimum alpha from the pointset...

Not sure if that (i.e. a minimum alpha) is possible as long as points are not regular? Is convex always achieved with "infinite" alpha?

Reply:
I agree not sure if it is possible to compute an "minimum alpha" for these datatypes... I will investigate a bit! But if there is not an easy solution, we just keep the function as is with the actual alpha needed as input 👍 

I am actually not sure what happens if you turn the alpha up very high.. but at least the alpha could be set to 0 to get convex results.

Reply:
Might have a solution to consider. It involves using scipy.spatial to finding the distances to the n'th nearest points and using the mean distance as the default alpha. Then the user could optimize the results with the alpha factor if needed. 

The operation seems very fast. I tested by creating random pointsets in rms and cut out a shape with a polygon (to make it more interesting 😄), this is the results:

![image](https://user-images.githubusercontent.com/61694854/235321455-57f93309-24bd-4e30-9773-3fa5d1867e88.png)


pasting in the code I used as example. Note the number of neighbors to compare distance is important for the results.. after testing I ended up with using the 30 closest points.

![image](https://user-images.githubusercontent.com/61694854/235322363-3e78c932-a721-4fbd-9450-4cc14b1bd189.png)

Reply:
Opposite results also (using the same pointsets but removing points outside the polygon).
![image](https://user-images.githubusercontent.com/61694854/235322546-e1db4309-411c-4eec-a732-b55ab222529c.png)

Reply:
Very interesting; I was not aware of the scipy functions. Will try to make something based on this!

Reply:
Implemented now; works well I think